applications/StructuralMechanicsApplication/custom_constitutive/linear_plane_stress.cpp

```diff
@@ -55,86 +55,136 @@ LinearPlaneStress::~LinearPlaneStress()
 {
-    CalculateMaterialResponsePK2(rValues);
+    CalculateMaterialResponsePK2(rParameterValues);
 }
 
-void LinearPlaneStress::CalculateMaterialResponseCauchy (Parameters& rValues)
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStress::CalculateMaterialResponseCauchy (Parameters& rParameterValues)
 {
-    CalculateMaterialResponsePK2(rValues);
+    CalculateMaterialResponsePK2(rParameterValues);
 }
 
-void LinearPlaneStress::FinalizeMaterialResponsePK1(Parameters& rValues)
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStress::FinalizeMaterialResponsePK1(Parameters& rParameterValues)
 {
     // TODO: Add if necessary
 }
 
-void LinearPlaneStress::FinalizeMaterialResponsePK2(Parameters& rValues)
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStress::FinalizeMaterialResponsePK2(Parameters& rParameterValues)
 {
     // TODO: Add if necessary
 }
 
-void LinearPlaneStress::FinalizeMaterialResponseCauchy(Parameters& rValues)
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStress::FinalizeMaterialResponseCauchy(Parameters& rParameterValues)
 {
     // TODO: Add if necessary
 }
 
-void LinearPlaneStress::FinalizeMaterialResponseKirchhoff(Parameters& rValues)
+//************************************************************************************
+//************************************************************************************
+
+void LinearPlaneStress::FinalizeMaterialResponseKirchhoff(Parameters& rParameterValues)
 {
     // TODO: Add if necessary
 }
 
+//************************************************************************************
+//************************************************************************************
+
+double& LinearPlaneStress::CalculateValue(Parameters& rParameterValues, const Variable<double>& rThisVariable, double& rValue)
+{
+    Flags &Options=rParameterValues.GetOptions();
+
+    const Properties& MaterialProperties  = rParameterValues.GetMaterialProperties();
+    Vector& StrainVector                  = rParameterValues.GetStrainVector();
+    Vector& StressVector                  = rParameterValues.GetStressVector();
+    const double& E          = MaterialProperties[YOUNG_MODULUS];
+    const double& NU    = MaterialProperties[POISSON_RATIO];
+    
+    if (rThisVariable == STRAIN_ENERGY)
+    {
+        if( Options.Is( ConstitutiveLaw::COMPUTE_STRAIN_ENERGY ) )
```

Comment:
you don't need to check this option. you already have the other if to control that ( i think that COMPUTE_STRAIN_ENERGY should fly out)